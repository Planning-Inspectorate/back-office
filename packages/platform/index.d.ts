import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
