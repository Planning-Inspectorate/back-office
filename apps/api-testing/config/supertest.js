// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
