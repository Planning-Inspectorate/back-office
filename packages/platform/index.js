export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
