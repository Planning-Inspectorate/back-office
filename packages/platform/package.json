{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
