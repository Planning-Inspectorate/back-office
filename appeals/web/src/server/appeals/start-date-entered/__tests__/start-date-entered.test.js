import { parseHtml } from '@pins/platform';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details';
const pageUrl = '/start-date-entered';

describe('appeal-details', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /start-date-entered', () => {
		it('should render the start date entered page', async () => {
			const response = await request.get(`${baseUrl}/1${pageUrl}`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});
