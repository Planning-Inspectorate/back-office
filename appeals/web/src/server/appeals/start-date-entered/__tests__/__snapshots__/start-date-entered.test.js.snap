// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appeal-details GET /start-date-entered should render the start date entered page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-panel govuk-panel--confirmation\\">
                <h1 class=\\"govuk-panel__title\\"> Time successfully published</h1>
                <div class=\\"govuk-panel__body\\">Case reference number
                    <br><strong>351062</strong>
                </div>
            </div>
            <p class=\\"govuk-body\\">The timetable is now created.</p>
            <h2>What happens next</h2>
            <p class=\\"govuk-body\\">We've sent the start letter email to the Appellant and LPA.</p>
            <p class=\\"govuk-body\\">The case has been published on the Appeals Casework Portal.</p>
            <p class=\\"govuk-body\\"><a class=\\"govuk-link\\" href=\\"/appeals-service/appeal-details/1\\">Go back to case details</a>
            </p>
        </div>
    </div>
</main>"
`;
